api/upload.py

```diff
@@ -73,16 +74,10 @@ def process_upload(request, strategy, container_type=None, id=None, origin=None)
 
     metadata = None
     if 'metadata' in form:
-        # Slight misnomer: the metadata field, if present, is sent as a normal form field, NOT a file form field.
-        metadata_file = form['metadata'].file
-        try:
-            metadata = json.loads(metadata_file.getvalue())
-        except AttributeError:
-            raise files.FileStoreException('wrong format for field "metadata"')
-
+        metadata = json.loads(form['metadata'].value)
```

Comment:
Please add the try except around that line.
I guess this is what makes the test to fail.